src/main/java/br/com/vaasschool/controller/SubcategoryController.java

```diff
@@ -0,0 +1,103 @@
+package br.com.vaasschool.controller;
+
+import br.com.vaasschool.controller.dto.SubcategoryDto;
+import br.com.vaasschool.controller.form.SubcategoryForm;
+import br.com.vaasschool.model.Category;
+import br.com.vaasschool.model.Subcategory;
+import br.com.vaasschool.repository.CategoryRepository;
+import br.com.vaasschool.repository.SubcategoryRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@Controller
+public class SubcategoryController {
+
+    private final CategoryRepository categoryRepository;
+    private final SubcategoryRepository subcategoryRepository;
+
+    public SubcategoryController(CategoryRepository categoryRepository, SubcategoryRepository subcategoryRepository) {
+        this.categoryRepository = categoryRepository;
+        this.subcategoryRepository = subcategoryRepository;
+    }
+
+    @GetMapping("/admin/subcategories/{categoryCode}")
+    public String listSubcategory(@PathVariable String categoryCode, Model model) {
+        Category category = categoryRepository.findByCode(categoryCode)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, categoryCode));
+        List<Subcategory> subcategories = subcategoryRepository.findAll();
+        List<SubcategoryDto> subCategoriesDtos = subcategories.stream().map(SubcategoryDto::new).toList();
+
+        model.addAttribute("subcategories", subCategoriesDtos);
+        model.addAttribute("categoryName", category.getName());
+        model.addAttribute("categoryCode", categoryCode);
+
+        return "subcategory/listSubcategory";
+    }
+
+    @GetMapping("/admin/subcategories/new")
+    public String showNew(SubcategoryForm subcategoryForm, Model model) {
+
+        List<Category> categories = categoryRepository.findAllByOrderByName();
+        model.addAttribute("categories", categories);
+        model.addAttribute("subcategoryForm", subcategoryForm == null ? new SubcategoryForm() : subcategoryForm);
+        return "subcategory/formSubcategory";
+    }
+
+    @PostMapping("/admin/subcategories")
```

Comment:
```suggestion 
    @Transactional
    @PostMapping("/admin/subcategories")
```

daí não precisa do save